compiler/code-gen/files/init-scripts.cpp

```diff
@@ -234,8 +235,11 @@ void CppMainFile::compile(CodeGenerator &W) const {
   W << OpenFile("main.cpp");
   W << ExternInclude("server/php-engine.h") << NL;
 
-  W << "int main(int argc, char *argv[]) " << BEGIN
-    << "return run_main(argc, argv, php_mode::" << G->settings().mode.get() << ")" << SemicolonAndNL{}
+  W << "int main(int argc, char *argv[]) " << BEGIN;
+  W << ShapeKeys::get_function_declaration() << ";" << NL;
+  W << ShapeKeys::get_function_name() << "();" << NL << NL;
+
+  W  << "return run_main(argc, argv, php_mode::" << G->settings().mode.get() << ")" << SemicolonAndNL{}
```

Comment:
It's very strange, that main() function is changed. I suppose, there is a more convenient way to propagate data from compilation to runtime

Reply:
Yea, true. I did so because of otherwise get weird error related with redefining malloc. Problem fixed with help of Denis.